File: getPersonJson/personGet.py
import json

# Función para obtener y mostrar la información de una persona por su ID
def obtener_persona_por_id(personas):
    # Solicitar ID al usuario
    try:
        id_usuario = int(input("Introduce el ID de la persona que deseas buscar: "))
    except ValueError:
        print("Por favor, ingresa un número válido para el ID.")
        return

    # Filtrar la persona correspondiente al ID
    persona = next((p for p in personas if p["id"] == id_usuario), None)

    if persona:
        # Mostrar la información de la persona
        print(persona)
    else:
        # Mensaje si no se encuentra la persona
        print(f"No se encontró una persona con ID {id_usuario}.")

# Leer el archivo JSON
def leer_archivo_json():
    try:
        with open('personas.json', 'r', encoding='utf-8') as file:
            personas = json.load(file)
            return personas
    except FileNotFoundError:
        print("El archivo personas.json no se encuentra.")
    except json.JSONDecodeError:
        print("Error al leer el archivo JSON, el formato no es válido.")
    return []

# Llamada principal
if __name__ == "__main__":
    personas = leer_archivo_json()
    if personas:
        print(f"Tamaño del objeto: {len(personas)} datos")
        obtener_persona_por_id(personas)

    else:
        print(f"sin datos del objeto")

# CMD: python  personGet.py

File: getPersonJson/personGet.php
<?php

// Función para obtener y mostrar la información de una persona por su ID
function obtenerPersonaPorID($personas) {
    // Mostrar el tamaño del objeto (cantidad de personas)
    echo "Tamaño del objeto: " . count($personas) . PHP_EOL;

    // Solicitar ID al usuario
    $idUsuario = readline("Introduce el ID de la persona que deseas buscar: ");

    // Convertir la entrada del usuario a un número
    $idNumero = (int) $idUsuario;

    // Filtrar la persona correspondiente al ID
    $persona = null;
    foreach ($personas as $p) {
        if ($p['id'] === $idNumero) {
            $persona = $p;
            break;
        }
    }

    if ($persona) {
        // Mostrar la información de la persona
        echo "Información de la persona con ID $idNumero:\n";
        echo "Nombre: " . $persona['nombre'] . " " . $persona['apellidos'] . "\n";
        echo "Edad: " . $persona['edad'] . "\n";
        echo "Ciudad de origen: " . $persona['ciudad_origen'] . "\n";
    } else {
        // Mensaje si no se encuentra la persona
        echo "No se encontró una persona con ID $idNumero.\n";
    }
}

// Leer el archivo JSON
function leerArchivoJson($archivo) {
    if (!file_exists($archivo)) {
        echo "El archivo $archivo no se encuentra.\n";
        return null;
    }

    $contenido = file_get_contents($archivo);

    // Decodificar el contenido JSON
    $personas = json_decode($contenido, true);

    if (json_last_error() !== JSON_ERROR_NONE) {
        echo "Error al leer el archivo JSON, el formato no es válido.\n";
        return null;
    }

    return $personas;
}

// Llamada principal
$personas = leerArchivoJson('personas.json');
if ($personas !== null) {
    obtenerPersonaPorID($personas);
}
?>

<!--  php personGet.php -->